auth/auth.go

```diff
@@ -105,19 +110,42 @@ func NewClient(ctx context.Context, c *internal.AuthConfig) (*Client, error) {
 			return nil, err
 		}
 	}
-
-	ks, err := newHTTPKeySource(ctx, googleCertURL, c.Opts...)
+	var hc *http.Client
+	if ctx != nil && len(c.Opts) > 0 {
```

Comment:
We should always initialize this. Otherwise we will end up passing nil to `internal.HTTPClient` below, which is not correct.